### ISSUE TITLE
Alerts being generated with wrong information

### ISSUE THREAD

**Issue by hobbitx** [prometheus/alertmanager#2877]
## General context:
When sending that an alarm was fixed, alertmanager is generating a different url than expected with wrong information about the query used. What we found was that the wrong field is the GeneratorURL, as the alert is redirected to the correct location, but with the wrong url.
We generated a link to take to an external dashboard, this generated link is wrong, but the other fields that use the same variables are correct.
The URL generated in this alarm belongs to another alarm. Both alarms have the same name, but the expr field is different between them.
An example of the alert that would be correct:
(Alert that rang and was resolved)
``` yml
- alert: kafka_lag_above_threshold
    expr: sum(kafka_burrow_partition_lag{group=~"cool-name-1", topic=~"DEADLETTER|REPLAY-DEADLETTER", topic!~"",prototype='sX'}) by (environment, prototype, group, topic, squad) > 50000 and on() ((hour() -3) >= 0 and (hour() -3) <= 24 and day_of_week() >= 0 and day_of_week() <= 6)
    for: 15m
    labels:
      severity: warning
      prototype: sX
    annotations:
      stack: '{{ $labels.environment }}-{{ $labels.prototype }}'
      value: '{{ $labels.topic }} @ {{ $labels.group }} has {{ $value }} messages
```
and the second alert:
``` yml
- alert: kafka_lag_above_threshold
    expr: sum(kafka_burrow_partition_lag{group=~"cool-name-2", topic=~".*", topic!~"",prototype='sX'}) by (environment, prototype, group, topic, squad) > 1000 and on() ((hour() -3) >= 9 and (hour() -3) <= 18 and day_of_week() >= 1 and day_of_week() <= 5)
    for: 10m
    labels:
      severity: warning
      prototype: sX
    annotations:
      stack: '{{ $labels.environment }}-{{ $labels.prototype }}'
      value: '{{ $labels.topic }} @ {{ $labels.group }} has {{ $value }} messages in lag'
```
In short the first alert rings, and when Alertmanager send the message that was corrected it comes with the link of the second alert.

Link send in the first alert (Alert that rang and was resolved):
``` 
https://url-base/graph?g0.expr=sum+by(environment,+prototype,+group,+topic,+squad)+(kafka_burrow_partition_lag{group=~"cool-name-2",prototype="sX",topic!~"",topic=~".*"})+>+1000+and+on()+((hour()+-+3)+>=+9+and+(hour()+-+3)+<=+18+and+day_of_week()+>=+1+and+day_of_week()+<=+5)&g0.tab=1
```
- Alert manager version: 0.23.0
- Prometheus Version: 2.32.1 

Can you help us understand what might be happening?
During our internal investigations we found nothing, there are even other alerts under the same conditions (same name and different expr) but this problem does not occur.

**Comment by 08d2** [comment]
Your two alerts have the same name, which is not valid. `promtool check rules <file.yaml>` would alert you to this problem.
